modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLViewEntriesDisplayContext.java

```diff
@@ -81,6 +86,13 @@ public DLViewEntriesDisplayContext(
 
 		_dlPortletInstanceSettingsHelper = new DLPortletInstanceSettingsHelper(
 			_dlRequestHelper);
+
+		_resourcePermissionLocalService =
+			(ResourcePermissionLocalService)liferayPortletRequest.getAttribute(
+				ResourcePermissionLocalService.class.getName());
+		_roleLocalService =
+			(RoleLocalService)liferayPortletRequest.getAttribute(
+				RoleLocalService.class.getName());
```

Comment:
Q: (suggestion for future changes, not for this PR) I don't like this way of injecting things; Ideally, dependencies required by a display context should be passed as constructor arguments. Now this cannot be done because the display context is initialized in a JSP, but maybe we could refactor things so that the initialization is done in the appropriate action class.